src/vnsw/agent/oper/interface.cc

```diff
@@ -461,23 +463,23 @@ void Interface::GetOsParams(Agent *agent) {
             "> querying flags for interface <" << name << "> " <<
             "Agent-index <" << id_ << ">");
         os_oper_state_ = false;
-        close(fd);
+        closesocket(fd);
         return;
     }
 
     os_oper_state_ = false;
     if ((ifr.ifr_flags & (IFF_UP | IFF_RUNNING)) == (IFF_UP | IFF_RUNNING)) {
         os_oper_state_ = true;
     }
-    close(fd);
+    closesocket(fd);
 
 #if defined(__linux__)
     mac_ = ifr.ifr_hwaddr;
 #elif defined(__FreeBSD__)
     mac_ = ifr.ifr_addr;
 #endif
 
-    int idx = if_nametoindex(name.c_str());
+    int idx = osspecific_if_nametoindex(name.c_str());
```

Comment:
As per our Tuesday discussion, we should avoid breaking compatibility if possible.

I believe simply replacing it with `if_nametoindex` would work okay. Am i wrong here?